databricks/koalas/tests/test_dataframe.py

```diff
@@ -2200,6 +2200,26 @@ def test_transpose(self):
                 repr(pdf3.transpose().sort_index()), repr(kdf3.transpose().sort_index())
             )
 
+    def _test_cumcount(self, pdf, kdf):
+        for column in pdf.columns:
+            self.assert_eq(pdf.groupby(column).cumcount(), kdf.groupby(column).cumcount())
+            self.assert_eq(
+                pdf.groupby(column).cumcount(ascending=False),
+                kdf.groupby(column).cumcount(ascending=False),
+            )
```

Comment:
Window functions won't guarantee the row order. we can just use `sort_index()`.

Reply:
Btw, these tests must be in `test_groupby` instead of `test_dataframe`.

Reply:
Could you also add tests for `SeriesGroupBy`? Please check the similar tests in `test_groupby`.

Reply:
I moved the tests and further extended it. Unfortunately, two parts are failing, where the groupby includes multiple columns and if one of those contains null values .. Couldn't find out why this is the case.

Do you have a pointer for `SeriesGroupBy`? I seem to only find DataFrame groupby tests there

Reply:
Something like `kdf.groupby(["b"])["a"].cumcount()` is calling `SeriesGroupBy`.

Reply:
Could you try with `almost=True` for the case containing null values in the result?

Reply:
The values are completely different and also `almost=True` does not work. What is wondering me, that already the groupby is yielding different results for koalas vs. pandas:
```
In [14]: kdf.groupby([("x", "a"), ("x", "b")]).first()                                                                                                                                 
Out[14]: 
               y
               c
(x, a) (x, b)   
1      NaN     4
4      NaN     1
1      20.0    2
       0.1     3

In [15]: pdf.groupby([("x", "a"), ("x", "b")]).first()                                                                                                                                 
Out[15]: 
               y
               c
(x, a) (x, b)   
1      0.1     3
       20.0    2
```

Reply:
ah, the case that groupkeys contain null values is a known issue #1007.
Could you work without null values in groupkeys here?